### ISSUE TITLE
Unable to use List(Str(attribute='')) in marshmallow 3.x

### ISSUE THREAD

**Issue by AbdealiLoKo** [marshmallow-code/marshmallow#1462]
This is somewhat of a follow up from a previous issue.
I have a case where I do:
```python
from marshmallow import fields, Schema
class MySchema(Schema):
    f = fields.List(fields.Str(attribute='name'), attribute='all_values')

print(MySchema().dumps({'all_values': [{'name': 1}]}))

```

This was working upto 3.0.0.rc5 and seems to not be giving me the same result after that.

In 3.0.0rc5: `{"f": ["1"]}` ------------- what I expected
In 3.0.0rc6: `{"f": ["{'name': 1}"]}`
In 3.0.0rc9: `{"f": ["{'name': 1}"]}`

In the release notes for 3.0.0.rc6, it says:
https://github.com/marshmallow-code/marshmallow/blob/3.2.2/CHANGELOG.rst#300rc6-2019-05-05
> Backwards-incompatible: Remove special-casing in fields.List and fields.Tuple for accessing nested attributes (:pr:`1188`). Use fields.List(fields.Pluck(...)) instead.

In the release notes for 3.0.0.rc9:
https://github.com/marshmallow-code/marshmallow/blob/3.2.2/CHANGELOG.rst#300rc9-2019-07-31
Haven't found what exactly caused the difference

In https://github.com/marshmallow-code/marshmallow/issues/1187#issuecomment-480615368 the alternative that is suggested it to use List(Pluck) like:
```python
from marshmallow import fields, Schema

class MySchema2(Schema):
    name = fields.String()

class MySchema(Schema):
    f = fields.List(fields.Pluck(MySchema2, 'name'), attribute='all_values')

print(MySchema().dumps({'all_values': [{'name': 1}]}))
```
But when I use this, I get:
In 3.0.0rc5: `{"f": ["1"]}` ------------- what I expected
In 3.0.0rc6: `{"f": ["1"]}` ------------- what I expected
In 3.0.0rc9: `{"f": ["1"]}` ------------- what I expected
In 3.2.2: `TypeError: list indices must be integers or slices, not str`

So, it looks like the alternative that @sloria mentioned in the comment is not valid anymore. And I'm not too clear on how to solve the serialization I require above using marshmallow

**Comment by AbdealiLoKo** [comment]
```python
from marshmallow import fields, Schema

class MySchema2(Schema):
    name = fields.String()
class MySchema(Schema):
    f = fields.Pluck(MySchema2, 'name', attribute='all_values', many=True)

print(MySchema().dumps({'all_values': [{'name': 1}]}))
```
This seems to work:
In 3.0.0rc5: `{"f": ["1"]}` ------------- what I expected
In 3.0.0rc6: `{"f": ["1"]}` ------------- what I expected
In 3.0.0rc9: `{"f": ["1"]}` ------------- what I expected
In 3.2.2: `{"f": ["1"]}` ------------- what I expected

**Comment by lafrech** [comment]
Thanks for reporting.

Using `Pluck` is indeed the way to go.

The issue with `Pluck` is discussed in #1424. There is a fix in #1425 that solves the error in https://github.com/marshmallow-code/marshmallow/issues/1462#issue-530938283.

The error also disappears if we remove the "Optimize dumping a list of Nested objects by calling dump(many=True)" lines.

Let's discuss this in #1424.
